From 84631da7ee3417fec4be072d89098f551e257ee2 Mon Sep 17 00:00:00 2001
From: Sachin Kamat <sachin.kamat@linaro.org>
Date: Tue, 4 Jun 2013 00:31:19 +0000
Subject: [PATCH 09/12] net: sun4i-emac: Remove redundant
 platform_set_drvdata()

Commit 0998d06310 (device-core: Ensure drvdata = NULL when no
driver is bound) removes the need to set driver data field to
NULL.

Signed-off-by: Sachin Kamat <sachin.kamat@linaro.org>
Cc: Stefan Roese <sr@denx.de>
Cc: Maxime Ripard <maxime.ripard@free-electrons.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit aff5c3557cf72fcad913d8b4b44b055a4e034bb5)
---
 drivers/net/ethernet/allwinner/sun4i-emac.c | 2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/ethernet/allwinner/sun4i-emac.c
+++ b/drivers/net/ethernet/allwinner/sun4i-emac.c
@@ -898,8 +898,6 @@ static int emac_remove(struct platform_d
 {
 	struct net_device *ndev = platform_get_drvdata(pdev);
 
-	platform_set_drvdata(pdev, NULL);
-
 	unregister_netdev(ndev);
 	free_netdev(ndev);
 
